### ISSUE TITLE
Will a tuner takes intermediate results into account?

### ISSUE THREAD

**Issue by zzzzzx-1115** [microsoft/nni#5332]
**Describe the issue**:
Recently I am trying early stopping algorithms built in NNI. However, I find ```report_final_result()``` seems not to work when a trial is early stopped, and the best value of intermediate results is not displayed in the web portal (Trial Details→Default metric) where final results of **succeeded** trials are shown. 

As the tuner works based on historical **final results**, I want to ask:  
1. Will the assessor report the "final result" which are inferred from already received immediate results (instead of getting it through ```report_final_result()``` conventionally) ? 
2. How does the tuner take advantages of those early-stopped (as well as user cancelled ones) trials?

**Comment by liuzhe-lz** [comment]
The early stopped trials are stopped before invoking `report_final_result()` so it is not supposed to work.
Each tuner may has its own strategy for handling early stopped trials. The default TPE tuner ignores those trials, as if they were never submitted.

We do have noticed that early stop is not well defined from the tuner's perspective.
We can add a config field like `tuner.early_stop_strategy: 'ignore' | 'latest'` if it is useful in your case.

**Comment by zzzzzx-1115** [comment]
Yes, I think a field like ```tuner.early_stop_strategy``` will be quite helpful. Look forward to such an enhancement! :-)
